### PR TITLE
Add markdownlinter as a pre-commit hook

### PR DESCRIPTION
**Pull request by arielger**
Add [markdownlint](https://github.com/DavidAnson/markdownlint) as a pre-commit hook using husky and lint-stagged libraries.

Now when you try to commit a markdown file with style errors, you will get something like this:

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,14 +5,24 @@
...
+  },
+  "lint-staged": {
+    "*.md": "npm run lint-markdown"
   },
   "engines": {
     "node": "4.5"
   },
   "devDependencies": {
     "commander": "^2.8.1",
+    "filehound": "^1.16.1",
```
Comment (ramasilveyra) [inline package.json]: Why you need `filehound` ? It's a peerdep?

File: package.json [hunk 2]
```diff
@@ -5,14 +5,24 @@
...
+    "husky": "^0.13.3",
     "imagemin": "^4.0.0",
-    "imagemin-optipng": "^4.3.0"
+    "imagemin-optipng": "^4.3.0",
+    "lint-staged": "^3.4.2",
+    "markdownlint": "^0.4.1",
+    "markdownlint-cli": "^0.3.1"
```
Comment (ramasilveyra) [inline package.json:24]: Maybe you don't need to list `markdownlint-cli` as devDep.

File: check-markdown.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+/* eslint-disable  no-console */
+const markdownlint = require('markdownlint');
+
+const options = {
+  files: process.argv.slice(2),
```
Comment (ramasilveyra) [inline check-markdown.js]: I think that you will loose some changed files, to properly support running this script from `lint-staged` and from npm script or directly you need to do something like this: https://github.com/auth0/auth0-website/blob/master/tools/scripts/prettier#L36-L54

To debug you can modify `lint-staged` source to see the stdout here https://github.com/okonet/lint-staged/blob/c7283b77c2cf3637e9f06f953b919a556ffde205/src/runScript.js#L19-L20 with:
```js
// ...
.then(result => {
	console.log(result.stdout);
// ...
```

**Comment by ramasilveyra** [comment]
With the last commit I believe that you can just use `markdownlint-cli`. 
> CLI argument --config is not mandatory. If it is not provided, markdownlint-cli looks for file .markdownlint.json in current folder, or for file .markdownlintrc in current or all upper folders. The algorithm is described in details on rc package page.

File: package.json (hunk 2 above)
Comment (arielger) [inline package.json]: I will try that ;)

**Pull request merged**